### ISSUE TITLE
feature request: artisan migrate should protect against parallel execute by default

### ISSUE THREAD

**Issue by tback** [laravel/framework#19615]
- Laravel Version: 5.3 (but 5.4 should be equally affected)
- PHP Version: 7.1.5
- Database Driver & Version: mysql

### Description:
Modern hosting environments run multiple instances of laravel for load-balancing and ha. They often execute migrations before launching the application itself. I've seen quite a few apps executing migrations in parallel by accident. This could be avoided if artisan migrate would lock in some way by default.
Maybe a table lock on the migration table would do the trick.

**Comment by themsaid** [comment]
For suggestions and feature requests please use the https://github.com/laravel/internals repo.

**Comment by tback** [comment]
Thanks, opened a new issue there: https://github.com/laravel/internals/issues/641
